src/NextGen/DonationForm/Models/DonationForm.php

```diff
@@ -0,0 +1,92 @@
+<?php
+
+namespace Give\NextGen\DonationForm\Models;
+
+use DateTime;
+use Give\Framework\FieldsAPI\Form;
+use Give\Framework\Models\Contracts\ModelCrud;
+use Give\Framework\Models\Contracts\ModelHasFactory;
+use Give\Framework\Models\Model;
+use Give\NextGen\DonationForm\Factories\DonationFormFactory;
+use Give\NextGen\DonationForm\Traits\DonationFormCRUD;
+use Give\NextGen\DonationForm\ValueObjects\DonationFormStatus;
+
+/**
+ * @unreleased
+ *
+ * @property int $id
+ * @property string $formTitle
+ * @property DateTime $createdAt
+ * @property DateTime $updatedAt
+ * @property DonationFormStatus $status
+ * @property array $settings
+ * @property string $blocksContent
+ * @property Form $schema
+ */
+class DonationForm extends Model implements ModelCrud, ModelHasFactory
+{
+    use DonationFormCRUD;
+
+    /**
+     * @inheritdoc
+     */
+    protected $properties = [
+        'id' => 'int',
+        'formTitle' => 'string',
+        'createdAt' => DateTime::class,
+        'updatedAt' => DateTime::class,
+        'status' => DonationFormStatus::class,
+        'settings' => 'array',
+        'blocksContent' => 'string',
+        'schema' => Form::class, // Read-only
+    ];
+
+    /**
+     * @unreleased
+     *
+     * @param string $settingName
+     *
+     * @return mixed
+     */
+    public function getSetting( $settingName )
+    {
+        return $this->settings[$settingName];
+    }
+
+    /**
+     * @unreleased
+     *
+     * @param $name
+     * @param $value
+     *
+     * @return $this
+     */
+    public function updateSetting($name, $value): self
+    {
+        $this->settings = array_merge($this->settings, [
+            $name => $value,
+        ]);
+        return $this;
+    }
```

Comment:
Although this is convenient, we identified the form settings as something we need to discuss further to determine its long-term functionality.  For now an array type works because it's flexible, but I'm afraid these getters and setters for this property will end up being deprecated as we move forward.  

As a side note, having getters and setters for array property types could be something we actually add to our core models like `$form->settings->get()` and `$form->settings->update()`.

So, until we finalize the functionality and data structure of form settings, let's remove these.

Reply:
Resolved by https://github.com/impress-org/givewp-next-gen/pull/63/commits/5122a3ce7bfe4044bab8c5b7ac1c3cd914df7340.